tests/test_settings.py

```diff
@@ -81,22 +81,23 @@ def _inner(**kwargs: str):
 
 safe_string = strategies.text(alphabet=string.ascii_letters + string.digits, min_size=1)
 nonempty_string_set = strategies.sets(safe_string, min_size=1)
-three_different_string_groups = strategies.tuples(
-    nonempty_string_set, nonempty_string_set, nonempty_string_set
-).filter(lambda ss: ss[0] != ss[1] and ss[0] != ss[2] and ss[1] != ss[2])
+four_different_string_groups = strategies.tuples(
+    *([nonempty_string_set] * 4),
```

Comment:
Are we sure this is not reusing the same [variable] pointer and duplicating it 4 times, thus resulting in an empty set after the filtering is executed?